### ISSUE TITLE
Max size of conditional variable?

### ISSUE THREAD

**Issue by MilesCranmer** [bayesiains/nflows#26]
Forgive my naivety, but is there a max size or required size of the conditional(/context) feature for a given normalizing flow? 

I have been trying to get a flow working with a large conditional feature vector, but am unable to get it working due to an apparent shape mismatch. 

In looking at one of the examples, I noticed that the number of conditional features/context features was equal to twice the number of input features. Is this required, or is there a different way to pass conditional information (without having to compute the full joint distribution over the extra features)?

I thus got my code working by projecting my current conditional feature to this smaller-sized space. Is this standard for autoregressive transform-based flows? Or is there a way to get the flow to work with very large conditional feature vectors?

Here is some example code I am working with:
```python
#Modeling a distribution in a space with `data_features` dimensions
# Thus, our context vector should have twice the dimensions:
hidden_features_transform = data_features * 2

# Create a projection matrix to encode the large conditional feature vector:
context_encoder = nn.Linear(conditional_features, hidden_features_transform)

# Give this encoder to the conditional base distribution
base_dist = ConditionalDiagonalNormal(
        shape=[data_features], 
        context_encoder=context_encoder)

# Create the flow with this many conditional features
transforms = []
for _ in range(num_layers):
    transforms.append(ReversePermutation(features=data_features))
    transforms.append(
        MaskedAffineAutoregressiveTransform(
            features=data_features,
            hidden_features=hidden_features_transform,
            context_features=conditional_features,
            num_blocks=5,
    ))
transform = CompositeTransform(transforms)
flow = Flow(transform, base_dist)
```
Currently, if I make this `hidden_features_transform` larger, I get shape mismatch errors, though I would expect this to just be another hidden layer hyperparameter I can change.

Thanks!
Miles

**Comment by MilesCranmer** [comment]
George Papamakarios answered my question; thanks! It was because I was using the ConditionalDiagonalNormal.

Answer:
- Nothing limiting conditional feature size in general for MAFs.
- Since the base distribution uses the conditional variable too, this limits the conditional variable here to the size of the base distribution's mean/variances.
- In general it is better to inject the conditional variable in the autoregressive transform rather than the base distribution.

So I think to fix this I'll just switch back the base distribution to standard normal
